File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-3.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode temp=head;
        int size=0;
        while(temp!=null){
            temp=temp.next;
            size++;
        }
        int target=size-n,count=0;
        temp=head;
        if(target==0) return head.next;
        while(temp!=null){
           if(count+1==target){
            temp.next=temp.next.next;
            }
            temp=temp.next;
            count++;
        }
        return head;
    }
}

File: Data Structures & Algorithms/reverse-nodes-in-k-group/submission-3.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def reverseKGroup(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        node=head
        for i in range(k):
            if not node:
                return head
            node=node.next
        
        prev,curr=None,head
        for i in range(k):
            temp=curr.next
            curr.next=prev
            prev=curr
            curr=temp
        
        head.next=self.reverseKGroup(curr,k)

        return prev